src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/BodyTagHelper.cs

```diff
@@ -19,9 +20,13 @@ public class BodyTagHelper : TagHelperComponentTagHelper
         /// </summary>
         /// <param name="manager">The <see cref="ITagHelperComponentManager"/> which contains the collection
         /// of <see cref="ITagHelperComponent"/>s.</param>
+        /// <param name="contextInitializer">The <see cref="ITagHelperComponentContextInitializer"/> initializes 
+        /// <see cref="ViewContext"/> for the <see cref="ITagHelperComponent"/>.</param>
         /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
-        public BodyTagHelper(ITagHelperComponentManager manager, ILoggerFactory loggerFactory)
```

Comment:
This is a breaking change

Reply:
Are default parameter values acceptable for public APIs? @dougbu

Reply:
We don't typically use default parameter values. Instead we prefer overloads. Here's a list of breaking change rules for future reference: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md

Reply:
It's more than "don't typically". Do not use default parameter values in `public` (or `protected`) methods, always add the overloads explicitly.

Reply:
I tried to do that before asking about default param values. `ActivatorUtilities` does not allow multiple applicable ctors. I'll need to change some of the design. https://github.com/aspnet/Mvc/issues/7330

Reply:
@pakrym does our DI system have a way to register custom factories (that use a different ctor) for certain types that `ActivatorUtilities.CreateFactory` will respect? 

Today we iterate through a list of TagHelper types and create them one by one with `ActivatorUtilities.CreateFactory(tagHelperType, Type.EmptyTypes)(serviceProvider, args)`

@jbagga if we can't do the above we can always work around the breaking change by querying for the component manager off of the `HttpContext.RequestServices`

Reply:
No, there is nothing like that right now.

Reply:
> No, there is nothing like that right now.

kk thought so after looking through the code. Thanks for confirming.